src/shared/credentials/userCredentialsUtils.ts

```diff
@@ -93,7 +93,10 @@ export class UserCredentialsUtils {
             throw new Error('Credentials file exists. Not overwriting it.')
         }
 
-        await writeFileAsync(this.getCredentialsFilename(), credentialsFileContents, 'utf8')
+        await writeFileAsync(this.getCredentialsFilename(), credentialsFileContents, {
+            encoding: 'utf8',
+            mode: 0o100600 // basic file (type 100) with 600 permissions
```

Comment:
Is it possible to use constants -- https://nodejs.org/api/fs.html#fs_fs_constants_1

Reply:
Note that on windows, only the 'write' flag is respected, and there is no distinction between owner/group/other. Can you add a comment to this effect?

Reply:
Also, do we need to set the first three octal digits? The FS should already be aware that it is a regular file.